### ISSUE TITLE
Can't build `github_graphql` when using `cargo clippy`

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#15671]
### Summary

I've been trying to run Clippy on https://github.com/rust-lang/triagebot, but it looks like `cargo clippy` gets stuck trying to build one of its internal crates, `github_graphql`. `cargo build` works just fine though.

Here are some of the things I've tried:
- `cargo clean`, a lot
- `cargo clippy --exclude=github_graphql --workspace`
- extract `github_graphql` to a separate Cargo project -- again, `build` works, `clippy` doesn't

Maybe it chokes on the build script? It is a bit involved: it generates queries based on a GraphQL schema, which could mean a lot of macro-derived code.

Also, my laptop isn't the most powerful out there, so maybe that's the actual bottleneck. Though it is usually able to compile even larger projects, _eventually_.

### Reproducer

I tried this:

```sh
git clone git@github.com:rust-lang/triagebot.git
cd triagebot
cargo clippy
```

I expected to see this happen:
Lints from all the packages or the workspace

Instead, this happened:
Lints only from `parser`, and then just this:
```
    Building [=======================> ] 375/383: github-graphql
```

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04) (built from a source tarball)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: x86_64-unknown-linux-gnu
release: 1.89.0
LLVM version: 20.1.8
```

**Comment by samueltardieu** [comment]
I think you need to be more patient. On my laptop (24 threads, 64GB RAM) it takes quite a long time: "Finished `dev` profile [unoptimized + debuginfo] target(s) in 4m 26s".

It would be interesting to find where the time is spent though.

**Comment by samueltardieu** [comment]
It looks most of the time is spent in the `four_forward_slashes` lint. Not running it cuts the Clippy run time to 38s.

**Comment by ada4a** [comment]
Oh wow, https://github.com/rust-lang/triagebot/pull/2180 made it _significantly_ faster! Thanks you:)
